strapi/strapi
# [Custom Fields] Improve error handling

### What does it do?

- Throws an error if there is a type mismatch between admin and server
- Checks the type is valid when registering a custom field in the admin
- Improves error message for type in the admin and on the server

### How to test it?

- Use two different types in the `mycustomfields` register function in the admin and on the server. For example `type: text` and `type: json` should throw an error when trying to save a custom field
- Try to use an invalid type in the admin or server register, for example `type: component` should throw an error when trying to start the app
